### PR TITLE
Feature/refactor personal code tests

### PR DESCRIPTION
**Pull request by clewis1**
Tests added to increase coverage.
https://companieshouse.atlassian.net/browse/IDVA3-1551

### CODE REVIEW

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

File: test/routers/handlers/personal-code/personCode.int.ts
[the remark this answers is not in the document]
Comment (clewis1) [reply test/routers/handlers/personal-code/personCode.int.ts:1]: Good spot

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

File: test/routers/handlers/personal-code/personalCode.int.ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+const mockPatchPscVerification = patchPscVerification as jest.Mock;
+
+jest.mock("../../../../src/services/pscService", () => ({
+    getPscIndividual: () => ({
+        httpStatusCode: HttpStatusCode.Ok,
+        resource: PSC_INDIVIDUAL
+    })
+}));
+
+describe("individual statement view", () => {
```
Comment (PhilipGlover) [inline test/routers/handlers/personal-code/personalCode.int.ts:30]: `individual statement view` ?

File: test/routers/personalCodeRouter.int.ts [hunk 2]
```diff
@@ -40,28 +35,8 @@ describe("personal code router tests", () => {
...
-        await request(app)
-            .post(getUrlWithTransactionIdAndSubmissionId(PrefixedUrls.PERSONAL_CODE, TRANSACTION_ID, PSC_VERIFICATION_ID))
-            .send({ pscType: "individual" })
-            .set({ "Content-Type": "application/json" })
-            .query({ companyNumber: COMPANY_NUMBER, lang: "en", pscType: "individual" })
-            .expect(HttpStatusCode.Found)
-            .expect("Location", expectedRedirectUrl);
+    it("Should render the personal code page with a successful status code", async () => {
+        const resp = await request(app).get(PrefixedUrls.PERSONAL_CODE);
+        expect(resp.status).toBe(200);
```
Comment (PhilipGlover) [inline test/routers/personalCodeRouter.int.ts:40]: ought there be a `(app).post` test in here too?  I know that this is done in the new handler tests, but I guess to be consistent it ought to be done here too.  pscTypeRouter has one.

File: test/routers/handlers/personal-code/personalCode.unit.ts [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+        const resp = await request(app).get(uri);
+
+        const $ = cheerio.load(resp.text);
+
+        expect(resp.status).toBe(HttpStatusCode.Ok);
+        expect($("a.govuk-back-link").attr("href")).toBe("/persons-with-significant-control-verification/transaction/11111-22222-33333/submission/662a0de6a2c6f9aead0f32ab/individual/psc-list?lang=en");
+    });
+
+    it("Should redirect to Individual Statement screen", async () => {
```
Comment (PhilipGlover) [inline test/routers/handlers/personal-code/personalCode.unit.ts:56]: ah I wonder if this test should be in the router test?  I'm a bit unsure of what type of tests should go where - router / handler, int / unit.

**Review by PhilipGlover — COMMENTED** [review]
Tests all pass and look good.  But perhaps a meeting of us all to agree what type of test goes where would be an idea?

File: test/routers/personalCodeRouter.int.ts (hunk 2 above)
Comment (PhilipGlover) [reply test/routers/personalCodeRouter.int.ts:40]: following dev discussion 11/6 we will wait for the routers and handlers to be refactored, and part of that will be re-organising the tests.  So my comment can be covered there.

File: test/routers/handlers/personal-code/personalCode.unit.ts (hunk 3 above)
Comment (PhilipGlover) [reply test/routers/handlers/personal-code/personalCode.unit.ts:56]: same as above - following dev discussion 11/6 we will wait for the routers and handlers to be refactored, and part of that will be re-organising the tests.  So my comment can be covered there.

**Comment by ch-code-analysis** [comment]
CI: Security warnings found!

**Review by PhilipGlover — APPROVED** [review]
required changes made and agreed to leave the location of tests as per the comments above

**Pull request merged**